src/components/Explore/TimeSelector.tsx

```diff
@@ -0,0 +1,177 @@
+  color: ${({ theme }) => theme.text2};
+  :hover {
+    color: ${({ theme }) => theme.text1};
+    cursor: pointer;
+    text-decoration: none;
+  }
+  > svg {
+    margin-right: 8px;
+  }
+`
+
+const InternalLinkMenuItem = styled(InternalMenuItem)`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: 0.5rem 0.5rem;
+  justify-content: space-between;
+  text-decoration: none;
+  :hover {
+    color: ${({ theme }) => theme.text1};
+    cursor: pointer;
+    text-decoration: none;
+  }
+`
+const MenuTimeFlyout = styled.span<{ flyoutAlignment?: FlyoutAlignment }>`
+  min-width: 150px;
+  max-height: 350px;
+  overflow: auto;
+  background-color: ${({ theme }) => theme.bg1};
+  box-shadow: 0px 0px 1px rgba(0, 0, 0, 0.01), 0px 4px 8px rgba(0, 0, 0, 0.04), 0px 16px 24px rgba(0, 0, 0, 0.04),
+    0px 24px 32px rgba(0, 0, 0, 0.01);
+  border: 1px solid ${({ theme }) => theme.bg0};
+  border-radius: 12px;
+  padding: 0.5rem;
+  display: flex;
+  flex-direction: column;
+  font-size: 16px;
+  position: absolute;
+  top: 3rem;
+  z-index: 100;
+
+  ${({ flyoutAlignment = FlyoutAlignment.RIGHT }) =>
+    flyoutAlignment === FlyoutAlignment.RIGHT
+      ? css`
+          right: 0rem;
+        `
+      : css`
+          left: 0rem;
+        `};
+  ${({ theme }) => theme.mediaWidth.upToMedium`
+    bottom: unset;
+    right: 0;
+    left: unset;
+  `};
+`
+
+const StyledMenuButton = styled.button`
+  width: 100%;
+  height: 100%;
+  border: none;
+  background-color: transparent;
+  margin: 0;
+  padding: 0;
+  height: 44px;
+  background-color: ${({ theme }) => theme.bg0};
+  border: 1px solid ${({ theme }) => theme.bg0};
+  padding: 0.15rem 0.5rem;
+  padding: 6px 12px 6px 12px;
+  border-radius: 12px;
+  font-size: 16px;
+  line-height: 24px;
+  font-weight: 400;
+
+  :hover,
+  :focus {
+    cursor: pointer;
+    outline: none;
+    border: 1px solid ${({ theme }) => theme.bg3};
+  }
+
+  svg {
+    margin-top: 2px;
+  }
+`
+
+const StyledMenu = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: relative;
+  border: none;
+  text-align: left;
+  width: 80px;
+`
+
+const StyledMenuContent = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  border: none;
+  width: 100%;
+  vertical-align: middle;
+  color: ${({ theme }) => theme.text1};
+
+  .chevron {
+    size: 15px;
+    color: ${({ theme }) => theme.text2};
+  }
+`
+
+const Chevron = styled.span`
+  color: ${({ theme }) => theme.text2};
+`
+
+// TODO: change this to reflect data pipeline
+export default function TimeSelector() {
+  const node = useRef<HTMLDivElement>()
+  const open = useModalOpen(ApplicationModal.TIME_SELECTOR)
+  const toggle = useToggleModal(ApplicationModal.TIME_SELECTOR)
+  const toggleMenu = useToggleModal(ApplicationModal.TIME_SELECTOR)
+  useOnClickOutside(node, open ? toggle : undefined)
+  const times = ['hour', 'day', 'week', 'month', 'year']
```

Comment:
I would have an enum for this
```TimeOption {
  Hour: 'hour',
  ...
}```

and then also have `TIMES = [TimeOption.Hour, ...` outside the component since it's a constant.